cadCAD/engine/simulation.py

```diff
@@ -114,7 +117,7 @@ def partial_state_update(
                 additional_objs
             ) -> List[Dict[str, Any]]:
 
-        last_in_obj: Dict[str, Any] = deepcopy(sL[-1])
+        last_in_obj: Dict[str, Any] = MappingProxyType(sL[-1])
```

Comment:
**Do not keep** line because it **is not** generally applicable to in sys models